compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -540,24 +540,27 @@ private void generateRecordValueCreateMethod(ClassWriter cw, List<BIRTypeDefinit
 
         mv.visitCode();
 
+        List<FullNamedTypeDef> recordTypeWithNames = mapToFullNames(recordTypeDefs);
+
         int fieldNameRegIndex = 1;
         Label defaultCaseLabel = new Label();
 
         // sort the fields before generating switch case
-        recordTypeDefs.sort(NAME_HASH_COMPARATOR);
+        recordTypeWithNames.sort(NAME_HASH_COMPARATOR);
 
-        List<Label> labels = createLabelsForSwitch(mv, fieldNameRegIndex, recordTypeDefs, defaultCaseLabel);
-        List<Label> targetLabels = createLabelsForEqualCheck(mv, fieldNameRegIndex, recordTypeDefs, labels,
+        List<Label> labels = createLabelsForSwitch(mv, fieldNameRegIndex, recordTypeWithNames, defaultCaseLabel);
+        List<Label> targetLabels = createLabelsForEqualCheck(mv, fieldNameRegIndex, recordTypeWithNames, labels,
                 defaultCaseLabel);
 
         int i = 0;
 
-        for (BIRTypeDefinition optionalTypeDef : recordTypeDefs) {
-            String fieldName = getTypeFieldName(optionalTypeDef.name.value);
+        for (FullNamedTypeDef typeDef : recordTypeWithNames) {
```

Comment:
Doesn't BIRTypeDefinition contain internal name? Why we need new list with mapToFullNames method